src/lib/errors.ts

```diff
@@ -0,0 +1,19 @@
+enum ErrorCode {
+  InvalidPublicKey   = 'INVALID_PUBLIC_KEY',
+  InvalidRedirectURL = 'INVALID_REDIRECT_URL',
+  InvalidService     = 'INVALID_SERVICE',
+  DataKeyNotFound    = 'DATA_KEY_NOT_FOUND'
+}
+  
+class ConnectError extends Error {
```

Comment:
I'm leaning towards `GandalfError`, connect seems like it could be anything - make the dev no go dey check their DB connection or whatever, wdyt?